nixos/modules/services/autotierfs.nix

```
@@ -0,0 +1,95 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.autotierfs;
+  ini = pkgs.formats.ini { };
+  format = lib.types.attrsOf ini.type;
+  stateDir = "/var/lib/autotier";
+
+  generateConfigName =
+    name: builtins.replaceStrings [ "/" ] [ "-" ] (lib.strings.removePrefix "/" name);
+  configFiles = builtins.mapAttrs (
+    name: val: ini.generate "${generateConfigName name}.conf" val
+  ) cfg.settings;
+
+  getMountDeps =
+    settings: builtins.concatStringsSep " " (builtins.catAttrs "Path" (builtins.attrValues settings));
+
+  mountPaths = builtins.attrNames cfg.settings;
+in
+{
+  options.services.autotierfs = {
+    enable = lib.mkEnableOption "the autotier passthrough tiering filesystem";
+    package = lib.mkPackageOption pkgs "autotier" { };
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType = format;
+      };
+      default = { };
+      description = ''
+        The contents of the configuration file for autotier.
+        See the [autotier repo](https://github.com/45Drives/autotier#configuration) for supported values.
+      '';
+      example = lib.literalExpression ''
+        {
+          "/mnt/autotier" = {
+            Global = {
+              "Log Level" = 1;
+              "Tier Period" = 1000;
+              "Copy Buffer Size" = "1 MiB";
+            };
+            "Tier 1" = {
+              Path = "/mnt/tier1";
+              Quota = "30GiB";
+            };
+            "Tier 2" = {
+              Path = "/mnt/tier2";
+              Quota = "200GiB";
+            };
+          };
+        }
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.enable -> cfg.settings != { };
```

```suggestion
        assertion = cfg.settings != { };
```
This assertion is dependent on the `mkIf cfg.enable` above already